core/include/detray/navigation/intersection/helix_plane_intersector.hpp

```diff
@@ -78,8 +78,9 @@ struct helix_intersector_impl<cartesian2D<algebra_t>, algebra_t> {
         const point3_type st = getter::vector<3>(sm, 0u, 3u);
 
         // Starting point on the helix for the Newton iteration
-        scalar_type s{
-            getter::norm(trf.point_to_global(mask.centroid()) - h.pos(0.f))};
+        scalar_type s{vector::dot(sn, mask.centroid() - h.pos(0.f)) /
```

Comment:
The mask centroid is in local cartesian coordinates, so you need:
```suggestion
        scalar_type s{vector::dot(sn, trf.point_to_global(mask.centroid()) - h.pos(0.f)) /
```

Reply:
OK